components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,365 @@
+    :return: 0 on success, -1 otherwise.
+    """
+
+    archive_ids: List[str]
+    logger.info("Starting to delete archives from the database.")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+
+            if dry_run:
+                logger.info("Running in dry-run mode. No changes will be made to the database.")
+
+            db_cursor.execute(query, params)
+            results = db_cursor.fetchall()
+
+            if 0 == len(results):
+                if "filter" == criteria:
+                    logger.info("No archives found within specified time frame.")
+                elif "ids" == criteria:
+                    logger.info("No archives found with matching IDs.")
+                return 0
+
+            archive_ids = [result["id"] for result in results]
+
+            if "ids" == criteria:
+                not_found_ids = set(params) - set(archive_ids)
+                if not_found_ids:
+                    logger.warning(
+                        f"""
+                        Failed to find archives with the following IDs:
+                        {', '.join(not_found_ids)}
+                        """
+                    )
+
+            id_string = ", ".join(f"'{archive_id}'" for archive_id in archive_ids)
+
+            db_cursor.execute(
+                f"""
+                DELETE FROM `{table_prefix}files`
+                WHERE archive_id in ({id_string})
+                """
+            )
+
+            db_cursor.execute(
+                f"""
+                DELETE FROM `{table_prefix}archive_tags`
+                WHERE archive_id in ({id_string})
+                """
+            )
+
+            if dry_run:
+                logger.info("Dry-run finished.")
+                db_conn.rollback()
+                return 0
+
+            db_conn.commit()
+            for archive_id in archive_ids:
+                logger.info(f"Deleted archive {archive_id} from the database.")
+
+    except Exception:
+        logger.exception("Failed to delete archives from the database. Aborting deletion.")
+        return -1
+
+    logger.info(f"Finished deleting archives from the database.")
+
+    for archive_id in archive_ids:
+        archive_path = archives_dir / archive_id
+        if not archive_path.is_dir():
+            logger.warning(f"Archive {archive_id} is not a directory. Skipping deletion.")
+            continue
+
+        logger.info(f"Deleting archive {archive_id} from disk.")
+        shutil.rmtree(archive_path)
+
+    logger.info(f"Finished deleting archives from disk.")
+
+    return 0
+
+
+def _delete_archives_by_filter(
```

Comment:
I personally feel those two functions are redundant.

If you make the query to be:
query = f"""
    DELETE FROM `{table}`
    WHERE {criteria}
    RETURNING id
"""

We should be able to construct the query inside _delete_archives?

The case where having these two functions will help is that if _delete_archives does not need to know the details of the query command. However, in your implementation, _delete_archives still needs to know the type (criteria) and the list of values for deleteion, so I would say either just embed query creation into _delete_archives, or refactor the code in some other way.